src/sleep/sleep.c

```diff
@@ -211,25 +212,13 @@ static int rtc_read_time(uint64_t *ret_sec) {
         return 0;
 }
 
-static int rtc_write_wake_alarm(uint64_t sec) {
-        char buf[DECIMAL_STR_MAX(uint64_t)];
-        int r;
-
-        xsprintf(buf, "%" PRIu64, sec);
-
-        r = write_string_file("/sys/class/rtc/rtc0/wakealarm", buf, WRITE_STRING_FILE_DISABLE_BUFFER);
-        if (r < 0)
-                return log_error_errno(r, "Failed to write '%s' to /sys/class/rtc/rtc0/wakealarm: %m", buf);
-
-        return 0;
-}
-
 static int execute_s2h(usec_t hibernate_delay_sec) {
 
         _cleanup_strv_free_ char **hibernate_modes = NULL, **hibernate_states = NULL,
                                  **suspend_modes = NULL, **suspend_states = NULL;
         usec_t original_time, wake_time, cmp_time;
-        int r;
+        int r, tfd;
```

Comment:
please use `_cleanup_close_` on the fd (an initialized to -1), so that it is automatically closed in error paths here.